pallets/token-factory/contract/contract.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v4.0/contracts/access/AccessControlEnumerable.sol";
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v4.0/contracts/utils/Context.sol";
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v4.0/contracts/token/ERC20/ERC20.sol";
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v4.0/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+
+/**
+ * @dev {ERC20} token, including:
+ *
+ *  - a minter role that allows for token minting (creation)
+ *  - a burner role that allows for token burning (deletion)
+ *
+ * This contract uses {AccessControl} to lock permissioned functions using the
+ * different roles - head to its documentation for details.
+ *
+ * The account that deploys the contract will be granted the minter and burner
+ * roles, as well as the default admin role, which will let it grant both minter
+ * and burner roles to other accounts.
+ */
+contract ERC20PresetMinterBurner is
+    Context,
+    AccessControlEnumerable,
+    ERC20Burnable
+{
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
```

Comment:
Do you have a scenario in mind for these roles being held by separate accounts?

Reply:
Nope, it'll just be sudo. We can simplify this contract by combining the roles and even getting rid of the fact that the permissions are delegatable.